src/neo4j_graphrag/experimental/pipeline/component.py

```diff
@@ -76,8 +82,14 @@ class Component(abc.ABC, metaclass=ComponentMeta):
     # added here for the type checker
     # DO NOT CHANGE
     component_inputs: dict[str, dict[str, str | bool]]
-    component_outputs: dict[str, dict[str, str | bool]]
+    component_outputs: dict[str, dict[str, str | bool | type]]
 
     @abc.abstractmethod
     async def run(self, *args: Any, **kwargs: Any) -> DataModel:
         pass
+
+    async def run_with_context(
+        self, context_: RunContext, *args: Any, **kwargs: Any
+    ) -> DataModel:
+        # default behavior to prevent a breaking change
```

Comment:
Could we add a docstring to let user knows why they should implement this?

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/301/commits/6a55af4f2f68122e9605da76c4bbd3b26ccb0379#diff-9a5dc912320769fa1bcd1821daa8c18bfc2b2c9355eb9855069a4582caace235R95